core/src/main/java/org/elasticsearch/common/io/PathUtils.java

```diff
@@ -93,6 +93,15 @@ public static Path get(Path[] roots, String path) {
     }
 
     /**
+     * Tries to resolve the given file uri against the list of available roots.
+     *
+     * If uri starts with one of the listed roots, it returned back by this method, otherwise null is returned.
+     */
+    public static Path get(Path[] roots, URI uri) {
+        return get(roots, Paths.get(uri).normalize().toString());
+    }
```

Comment:
I think this needs to call PathUtils.get (not Paths.get) so that it does not use the JVM default filesystem, in the case one is set differently in tests.